src/core/src/op/scatter_update.cpp

```diff
@@ -124,3 +124,65 @@ bool op::v3::ScatterUpdate::has_evaluate() const {
     }
     return false;
 }
+
+namespace {
+OPENVINO_SUPPRESS_DEPRECATED_START
+bool scatter_label_evaluator(const Node* node, TensorLabelVector& output_labels) {
+    const auto& input_values = node->input_values();
+    if (input_values.size() < 4) {
+        return false;
+    }
+
+    constexpr auto data_in_idx = 0;
+    constexpr auto updates_in_idx = 2;
+    bool in0_has_no_labels = ov::has_no_labels(input_values[data_in_idx].get_tensor().get_value_label());
+    bool in2_has_no_labels = ov::has_no_labels(input_values[updates_in_idx].get_tensor().get_value_label());
+
+    if (in0_has_no_labels && in2_has_no_labels) {
+        return false;
+    }
+
+    HostTensorVector input_tensors(input_values.size());
+    for (size_t i = 0; i < input_values.size(); ++i) {
+        const auto& input = input_values[i];
+        if (i == data_in_idx || i == updates_in_idx) {
+            auto input_labels = input.get_tensor().get_value_label();
```

Comment:
It looks like get labels and check `no label` for input 0, 2 is done twice (previous at L:138,139). Can be reduced to do this once per input?

Reply:
Done